src/components/UI/ToggleSwitch/ToggleSwitch.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { FunctionComponent } from "react";
+
+export interface ToggleProps {
+  isOn: boolean;
+  handleToggle?: () => void;
+}
+/**
+ * ToggleSwitch is a controlled component, depending on the state it is in,
+ * it will render the on or off text accordingly
+ * @param isOn state of toggle
+ * @param handleToggle toggle handler, a callback that invokes if the state
+ * changes
+ */
+export const ToggleSwitch: FunctionComponent<ToggleProps> = ({ isOn, handleToggle }) => {
+  return (
+    <label data-testid="toggle-switch-label" className="flex items-center cursor-pointer">
+      <div className="relative">
+        <input
+          checked={isOn}
+          className="hidden"
+          onChange={handleToggle}
+          type="checkbox"
+          id="toggle-switch"
```

Comment:
this id still need?

Reply:
we are selecting with data-testid only now? (integration tests)
if so then id wont be needed

Reply:
um not sure about current state, was thinking yeah probably stick to testid for tests <- explicit that this is used in test so kinda know meddle with care

normal id could be use for anchor `href` / label `for` kinda stuffs, thought it's good that we don't mix them

Reply:
👌